src/App.js

```diff
@@ -13,10 +14,105 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods 1</h1>
-  </div>
-);
+class App extends React.Component {
+  state = {
+    isStarted: false,
+    visibleGoods: [...goodsFromServer],
+    valueOfSelect: '1',
+  };
+
+  start = () => {
+    this.setState({ isStarted: true });
+  };
+
+  reset = () => {
+    this.setState({
+      visibleGoods: [...goodsFromServer],
+      valueOfSelect: '1',
+    });
+  };
+
+  reverse = () => {
+    this.setState(
+      state => ({ visibleGoods: [...state.visibleGoods].reverse() })
+    );
+  };
+
+  sortByName = () => {
+    this.setState(
+      state => ({
+        visibleGoods: [...state.visibleGoods]
+          .sort((a, b) => a.localeCompare(b)),
+      }),
+    );
+  };
+
+  sortByLength = () => {
+    this.setState(
+      state => ({
+        visibleGoods: [...state.visibleGoods]
+          .sort((a, b) => a.length - b.length),
+      }),
+    );
+  };
+
+  selectByLength = (event) => {
+    const value = +event.target.value;
+
+    this.setState(
+      state => ({
+        visibleGoods: [...goodsFromServer]
```

Comment:
```suggestion
        visibleGoods: goodsFromServer
```
копія не потрібна, filter і так робить копію